CRM/Contribute/Form/SoftCredit.php

```diff
@@ -52,12 +52,6 @@ public static function preProcess(&$form) {
         'entity_id' => $contriDAO->contribution_page_id,
       );
       $profileId = CRM_Core_BAO_UFJoin::getUFGroupIds($ufJoinParams);
-
-      //check if any honree profile is enabled if yes then assign its profile type to $_honoreeProfileType
-      // which will be used to constraint soft-credit contact type in formRule, CRM-13981
-      if (!empty($profileId[0]) && !empty($profileId[2])) {
-        $form->_honoreeProfileType = CRM_Core_BAO_UFGroup::getContactType($profileId[0]);
-      }
```

Comment:
Further, 

1. I think we should also remove any reference to this preProcess() to avoid any fatal errors caused by this change - one I can see is [here](https://github.com/civicrm/civicrm-core/blob/master/CRM/Contribute/Form/Contribution.php#L305-L307).

2. Remove [definition of `_honoreeProfileType`](https://github.com/civicrm/civicrm-core/blob/master/CRM/Contribute/Form/Contribution.php#L157) as it doesn't seem to be used elsewhere.

       jitendra$ grep -irn honoreeProfileType CRM
       CRM/Contribute/Form/Contribution.php:157:  public $_honoreeProfileType;